sqlite/lib/SQLiteService.js

```diff
@@ -15,6 +15,19 @@ class SQLiteService extends SQLService {
         dbc.function('session_context', key => dbc[$session][key])
         dbc.function('regexp', { deterministic: true }, (re, x) => (RegExp(re).test(x) ? 1 : 0))
         dbc.function('ISO', { deterministic: true }, d => d && new Date(d).toISOString())
+
+        // define date and time functions in js to allow for throwing errors
+        const isTime = /^\d{1,2}:\d{1,2}:\d{1,2}$/
+        const hasTimezone = /([+-]\d{1,2}:?\d{0,2}|Z)$/
+        const throwNaN = n => { if (Number.isNaN(n)) { throw new Error(`Value does not contain a date "${n}"`) } else { return n } }
```

Comment:
I don't think that this gives a good error message. If I provide an invalid time or date, it'll end up in an error message `Value does not contain a date "NaN"`. it's not clear whether the user input was a time or what was the invalid value.

Reply:
So we should differentiate between year, month, day (which require a date) and hour, minute second (which require a time)? @BobdenOs do you agree?

Reply:
The error does not need to be related to the function trying to parse the value as the value is only `NaN` if the original value is an invalid `Date`. So even before it would be possible to apply the extraction it already failed.

Reply:
Yes, but now we have the possibility to raise a better error message, don't we?